subsys/net/ip/route.c

```diff
@@ -595,12 +595,12 @@ struct in6_addr *net_route_get_nexthop(struct net_route_entry *route)
 		}
 
 		ipv6_nbr_data = net_ipv6_nbr_data(nexthop_route->nbr);
-		NET_ASSERT(ipv6_nbr_data);
+		if (ipv6_nbr_data) {
+			addr = &ipv6_nbr_data->addr;
+			NET_ASSERT(addr);
```

Comment:
Shouldn't assert of addr also be replaced with a "if" check?

Reply:
In this case it would not make a difference as the function can return NULL just fine.